engines/unknown-game-dialog.cpp

```diff
@@ -0,0 +1,133 @@
+	int buttonHeight = g_gui.xmlEval()->getVar("Globals.Button.Height", 0);
+
+	//Calculate the size the dialog needs
+	Common::Array<Common::String> lines;
+	int maxlineWidth = g_gui.getFont().wordWrapText(_reportTranslated, screenW - 2 * 20, lines);
+
+	_w = MAX(MAX(maxlineWidth, 0), (2 * buttonWidth) + 10) + 20;
+	int lineCount = lines.size() + 1;
+
+	_h =  3 * kLineHeight;
+	_h += lineCount * kLineHeight;
+
+	// Center the dialog
+	_x = (screenW - _w) / 2;
+	_y = (screenH - _h) / 2;
+
+	// Buttons
+	int closeButtonWidth = MAX(buttonWidth, g_gui.getFont().getStringWidth(_("Close")) + 10);
+	int copyToClipboardButtonWidth = MAX(buttonWidth, g_gui.getFont().getStringWidth(_("Copy to clipboard")) + 10);
+	int openBugtrackerURLButtonWidth = MAX(buttonWidth, g_gui.getFont().getStringWidth(_("Report game")) + 10);
+	int closeButtonPos = _w - closeButtonWidth - 10;
+	int copyToClipboardButtonPos = _w - copyToClipboardButtonWidth - closeButtonWidth - 15;
+	int openBugtrackerURLButtonPos = _w - copyToClipboardButtonWidth - closeButtonWidth - openBugtrackerURLButtonWidth - 20;
+	_closeButton = new ButtonWidget(this, closeButtonPos, _h - buttonHeight - 8, buttonWidth, buttonHeight, _("Close"), 0, kClose);
+
+	//Check if we have clipboard functionality
+	if (g_system->hasFeature(OSystem::kFeatureClipboardSupport)) {
+	   new ButtonWidget(this, copyToClipboardButtonPos, _h - buttonHeight - 8, copyToClipboardButtonWidth, buttonHeight, _("Copy to clipboard"), 0, kCopyToClipboard);
+	}
+
+	//Check if we have support for opening URLs
+	if (g_system->hasFeature(OSystem::kFeatureOpenUrl)) {
+		new ButtonWidget(this, openBugtrackerURLButtonPos, _h - buttonHeight - 8, openBugtrackerURLButtonWidth, buttonHeight, _("Report game"), 0, kOpenBugtrackerURL);
+		//Formatting the reportData for bugtracker submission [replace line breaks]...
+		_bugtrackerGameData = _reportData;
+		while (_bugtrackerGameData.contains("\n")) {
+			replace(_bugtrackerGameData, "\n", "%0A");
+		}
+	}
+
+	// Each line is represented by one static text item.
+	uint y = 10;
+	for (uint i = 0; i < lines.size(); i++) {
+		new StaticTextWidget(this, 10, y, _w, kLineHeight,
+								lines[i], Graphics::kTextAlignLeft);
+		y += kLineHeight;
+	}
+
+	y += kLineHeight;
+}
+
+void UnknownGameDialog::handleCommand(CommandSender *sender, uint32 cmd, uint32 data) {
+	switch(cmd) {
+	case kCopyToClipboard:
+	{
+		g_system->setTextInClipboard(_reportData);
+		MessageDialog dialog(_("All necessary information about your game has been copied into the clipboard"), _("OK"));
+		dialog.runModal();
+		break;
+	}
+	case kClose:
+	{
+		close();
+		break;
+	}
+	case kOpenBugtrackerURL:
+	{
+		g_system->openUrl("https://bugs.scummvm.org/newticket?summary=[UNK] Unknown game for engine " + _bugtrackerAffectedEngine + ": &description=" + _bugtrackerGameData + "&component=Engine%3A+"+_bugtrackerAffectedEngine+"&type=enhancement"+"&keywords=unknown-game "+_bugtrackerAffectedEngine);

```

Comment:
Using `String::format()` would make this a bit more readable.

Reply:
Fixed!